tests/console/check_package_version.pm

```diff
@@ -45,35 +47,34 @@ sub cmp_version {
 }
 
 sub cmp_packages {
-    my ($pcks, $pckv) = @_;
+    my ($pcks, $pckv, $jsc) = @_;
     record_info($pcks, "$pcks version check after migration");
     my $output = script_output("zypper se -s $pcks | grep -w $pcks | head -1 | awk -F '|' '{print \$4}'", 80);
     if ($output ne '' && !cmp_version($pckv, $output)) {
-        record_info("Version Failed", "The $pcks version is $output, but request is $pckv");
-        return $pcks;
+        record_soft_failure("$jsc, The $pcks version is $output, but request is $pckv");
     }
-    return $pcks if ($output eq '');
+    record_soft_failure("$jsc, The $pcks is not existed") if ($output eq '');
 }
 
 sub run {
 
     select_console 'root-console';
 
-    my @failed_pcks;
     foreach my $key (keys %package) {
-        my $pcks = cmp_packages($key, $package{$key});
-        push @failed_pcks, $pcks if $pcks;
+        my $pcks = cmp_packages($key, $package{$key}[0], $package{$key}[1]);
     }
 
     if (get_var('ARCH') =~ /s390x/) {
         foreach my $key (keys %package_s390x) {
-            my $pcks = cmp_packages($key, $package_s390x{$key});
-            push @failed_pcks, $pcks if $pcks;
+            my $pcks = cmp_packages($key, $package_s390x{$key}[0], $package_s390x{$key}[1]);
         }
     }
 
     assert_script_run('python3 --version', 30);
```

Comment:
Please add Jira No. as comments or viewer will not know why run 'python3 -- version' here.